x-pack/plugins/actions/server/builtin_action_types/servicenow/schema.ts

```diff
@@ -10,6 +10,10 @@ import { DEFAULT_ALERTS_GROUPING_KEY } from './config';
 
 export const ExternalIncidentServiceConfigurationBase = {
   apiUrl: schema.string(),
+  isOAuth: schema.boolean({ defaultValue: false }),
+  userIdentifierValue: schema.nullable(schema.string()), // required if isOAuth = true
```

Comment:
I think the `kbn-schema` supports conditional but maybe it is not that straightforward.